src/overlays/actors/ovl_Door_Warp1/z_door_warp1.c

```diff
@@ -22,77 +45,1005 @@ const ActorInit Door_Warp1_InitVars = {
+
+    if (this->warpTimer == 0) {
+        Math_SmoothStepToF(&this->crystalAlpha, 0.0f, 0.1f, 4.0f, 1.0f);
+        if (this->crystalAlpha <= 150.0f) {
+            player->actor.gravity = -0.1f;
+        }
+        if (this->crystalAlpha <= 0.0f) {
+            DoorWarp1_SetupAction(this, DoorWarp1_FloatPlayer);
+        }
+    } else {
+        this->warpTimer--;
+    }
+    this->actor.shape.rot.y += 0x320;
+}
+
+void DoorWarp1_FloatPlayer(DoorWarp1* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    player->actor.gravity = -0.1f;
+}
+
+void DoorWarp1_PurpleCrystal(DoorWarp1* this, GlobalContext* globalCtx) {
+    if (this->warpTimer != 0) {
+        this->warpTimer--;
+        Math_SmoothStepToF(&this->actor.scale.x, 0.0499f, 0.2f, 0.05f, 0.001f);
+        Math_SmoothStepToF(&this->actor.scale.y, 0.077f, 0.2f, 0.05f, 0.001f);
+        Math_SmoothStepToF(&this->actor.scale.z, 0.09f, 0.2f, 0.05f, 0.001f);
+        Math_SmoothStepToF(&this->crystalAlpha, 255.0f, 0.2f, 5.0f, 0.1f);
+    }
+}
+
+void DoorWarp1_ChooseInitialAction(DoorWarp1* this, GlobalContext* globalCtx) {
+    switch (this->actor.params) {
+        case WARP_DUNGEON_CHILD:
+        case WARP_CLEAR_FLAG:
+        case WARP_SAGES:
+        case WARP_YELLOW:
+        case WARP_BLUE_RUTO:
+        case WARP_DESTINATION:
+        case WARP_UNK_7:
+        case WARP_ORANGE:
+        case WARP_GREEN:
+        case WARP_RED:
+            DoorWarp1_SetupWarp(this, globalCtx);
+            break;
+        case WARP_DUNGEON_ADULT:
+            DoorWarp1_SetupAdultDungeonWarp(this, globalCtx);
+            break;
+        case WARP_BLUE_CRYSTAL:
+            DoorWarp1_SetupBlueCrystal(this, globalCtx);
+            break;
+        case WARP_PURPLE_CRYSTAL:
+            DoorWarp1_SetupPurpleCrystal(this, globalCtx);
+            break;
+    }
+}
+
+void DoorWarp1_AwaitClearFlag(DoorWarp1* this, GlobalContext* globalCtx) {
+    if (Flags_GetTempClear(globalCtx, this->actor.room)) {
+        this->warpTimer = 200;
+        Audio_QueueSeqCmd(0x21);
+        DoorWarp1_SetupAction(this, func_809995D4);
+    }
+}
+
+void func_809995D4(DoorWarp1* this, GlobalContext* globalCtx) {
+    if (this->warpTimer == 0) {
+        if (this->actor.xzDistToPlayer < 100.0f) {
+            this->actor.world.pos.x = -98.0f;
+            this->actor.world.pos.y = 827.0f;
+            this->actor.world.pos.z = -3228.0f;
+        }
+        Lights_PointNoGlowSetInfo(&this->upperLightInfo, this->actor.world.pos.x, this->actor.world.pos.y,
+                                  this->actor.world.pos.z, 200, 255, 255, 255);
+        Lights_PointNoGlowSetInfo(&this->lowerLightInfo, this->actor.world.pos.x, this->actor.world.pos.y,
+                                  this->actor.world.pos.z, 200, 255, 255, 255);
+        DoorWarp1_SetupAction(this, DoorWarp1_ActivateWarp);
+    }
+    this->warpTimer--;
+}
+
+void DoorWarp1_ActivateWarp(DoorWarp1* this, GlobalContext* globalCtx) {
```

Comment:
I'm not sure this function is named right, I would think "activate" means Link stepped into the warp but that's not what this does right? this is more like `DoorWarp1_Appear` / FadeIn ? not sure

Reply:
To me I think the warp going from a state of unusable to usable is describable as it "activating", but as you pointed out the player stepping on it can also be considered such. I like "Appear" though so I've changed it to that, which is much less ambiguous.